src/main/java/org/folio/rest/impl/base/AbstractEmail.java

```diff
@@ -202,26 +200,49 @@ protected Future<Configurations> lookupConfiguration(Map<String, String> request
           return succeededFuture(config);
         }
         String errorMessage = String.format(ERROR_LOOKING_UP_MOD_CONFIG,
-          url, response.statusCode(), response.bodyAsString());
+          path, response.statusCode(), response.bodyAsString());
         logger.error(errorMessage);
         return failedFuture(new ConfigurationException(errorMessage));
       });
   }
 
-  protected Future<Configurations> validateConfiguration(Configurations configurations) {
-    if (isIncorrectSmtpServerConfig(configurations)) {
-      String errorMessage = String.format(ERROR_MIN_REQUIREMENT_MOD_CONFIG,
-        REQUIREMENTS_CONFIG_SET);
-      logger.error(errorMessage);
-      return failedFuture(new SmtpConfigurationException(errorMessage));
-    }
+  private Future<SmtpConfiguration> copyConfigurationAndDeleteFromModConfig(
+    Configurations configurations, OkapiClient okapiClient) {
 
-    return succeededFuture(configurations);
+    return succeededFuture(configurations)
+      .map(EmailUtils::convertSmtpConfiguration)
+      .compose(EmailUtils::validateSmtpConfiguration)
+      .compose(smtpConfigurationService::createSmtpConfiguration)
+      .onSuccess(smtpConfig -> deleteEntriesFromModConfig(configurations, okapiClient));
+  }
+
+  private void deleteEntriesFromModConfig(Configurations configurationsToDelete,
+    OkapiClient okapiClient) {
+
+    logger.warn("Removing configuration from mod-config");
+
+    configurationsToDelete.getConfigs().stream()
+      .map(Config::getId)
+      .forEach(id -> {
+        String path = format(DELETE_CONFIG_PATH_TEMPLATE, CONFIG_BASE_PATH, id);
+
+        okapiClient.deleteAbs(path)
+          .send()
+          .compose(response -> {
+            if (response.statusCode() == HTTP_NO_CONTENT.toInt()) {
+              logger.info("Successfully deleted configuration entry {}", id);
+              return succeededFuture();
+            }
+            String errorMessage = format("Failed to delete configuration entry %s", id);
+            logger.error(errorMessage);
+            return failedFuture(new ConfigurationException(errorMessage));
```

Comment:
No need to return anything, the result will be ignored anyway. And we can use `onComplete` or `onSuccess`+`onFailure` instead of `compose`.